Fix for acceptable use to handle case where user mustAccept

hdeadman opened the pull request:
Closes #2464

It turns out that acceptable use wasn't working at all in 5.1, as far as I could tell, just using the default repository (none). It didn't have to do with using it in combination with x509. If you needed to accept the flow was just sending you back to the login page. 

Should I be trying to share the "mustAccept" string constant across the two classes where it is used? I added a new transition for "mustAccept" but should I have just changed the default transition? 

I would still like to have the option of acceptable use popping-up when users are doing x509 authentication but I can open another issue for that since that is more of a new feature.

mmoayyed commented:
Excellent, thank you for tracking this down.

Two things:

- I very much like your suggestion of sharing the `mustAccept` constant across classes. Please do. Also I am in favor of the new explicit transition, and would prefer that you get rid of the default in fact, if wouldn't cause any other issues. I should also add that this can perfectly be merged as-is, if you decided to forego any of those changes.
- As x509 authentication and AUP, yes please open up a separate issue or better yet, skip the issue and just go to the pull request directly. No need for extra "paperwork".

hdeadman commented:
I made the updates and squashed the commit. I wasn't sure if the default flow would kick in if there was an unhandled exception but it's probably better to let that go to some global error handler anyway, so I removed it.
